url/Slug.ts

```diff
@@ -1,27 +1,14 @@
-
-
-export default class Slug
+export class Slug
 {
+    private readonly transforms : Array<[RegExp, string]>;
     /**
-     * Constructs a new instance
-     *
-     * @param {Array.<Array.<*>>} transforms
-     * @param {?RegExp} sanitize
+     * The sanitizing regexp, that removes all unwanted characters
      */
-    constructor (transforms = [], sanitize = null)
+    private readonly sanitize : RegExp;
+
+    constructor (transforms : Array<[RegExp, string]> = [], sanitize : null|RegExp= null)
```

Comment:
This type is actually wrong, and explains why you were able to remove the `flags` check further below.

The correct type is: `Array<[RegExp|string, string]>`
Also: please add a named type for that in this file.